yt_dlp/networking/_requests.py

```diff
@@ -180,14 +182,27 @@ def proxy_manager_for(self, proxy, **proxy_kwargs):
             extra_kwargs['proxy_ssl_context'] = self._proxy_ssl_context
         return super().proxy_manager_for(proxy, **proxy_kwargs, **self._pm_args, **extra_kwargs)
 
+    # Skip `requests` internal verification; we use our own SSLContext
+    # requests 2.31.0+
     def cert_verify(*args, **kwargs):
-        # Lean on our SSLContext for cert verification
         pass
 
+    # requests 2.31.0-2.32.1
     def _get_connection(self, request, *_, proxies=None, **__):
-        # Lean on our SSLContext for cert verification
         return self.get_connection(request.url, proxies)
 
+    # requests 2.32.2+: Reimplementation without `_urllib3_request_context`
+    def get_connection_with_tls_context(self, request, verify, proxies=None, cert=None):
+        if proxy := requests.utils.select_proxy(request.url, proxies):
+            proxy = requests.utils.prepend_scheme_if_needed(proxy, 'http')
+            if not urllib3.util.parse_url(proxy).host:
+                raise requests.exceptions.InvalidProxyURL(
+                    'Please check proxy URL. It is malformed and could be missing the host.')
```

Comment:
Technically we already validate all this already, so only need to use our select_proxy

Up to you though

Reply:
```py
    def get_connection_with_tls_context(self, request, verify, proxies=None, cert=None):
        manager = self.poolmanager
        if proxy := select_proxy(request.url, proxies):
            manager = self.proxy_manager_for(proxy)

        return manager.connection_from_url(request.url)
```
this passes all the tests; do we have to normalize the url though?

Reply:
The request handler validation requires the proxy URL has a scheme

Reply:
do we need to normalize it with `urllib3.util.parse_url(request.url).url` though?

Reply:
Shouldn't need to but if requests does that ig it doesn't hurt